wlauto/workloads/googlephotos/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -145,55 +137,260 @@ private void gesturesTest() throws Exception {
 
-        // Select first photograph
-        selectPhoto(0);
+        Iterator<Entry<String, SeekBarTestParams>> it = testParams.entrySet().iterator();
+
+        // Select second photograph
+        selectPhoto(1);
         UiObject editView = getUiObjectByResourceId("com.google.android.apps.photos:id/edit",
                                                     "android.widget.ImageView");
         editView.click();
 
-        UiObject editColor = getUiObjectByText("Colour", "android.widget.RadioButton");
-        editColor.click();
+        // Manage potential different spelling of UI element
+        UiObject editColor = new UiObject(new UiSelector().text("Color"));
+        UiObject editColour = new UiObject(new UiSelector().text("Colour"));
+
+        if (editColor.exists()) {
+            editColor.click();
+        } else if (editColour.exists()) {
+            editColour.click();
+        } else {
+            throw new UiObjectNotFoundException(String.format("Could not find \"%s\" \"%s\"",
+                                                              "Color/Colour", "android.widget.RadioButton"));
+        }
 
         UiObject seekBar = getUiObjectByResourceId("com.google.android.apps.photos:id/cpe_strength_seek_bar",
                                                    "android.widget.SeekBar");
-        seekBar.swipeLeft(10);
 
-        UiObject accept = getUiObjectByDescription("Accept", "android.widget.ImageView");
-        accept.click();
+        String viewName = "com.google.android.apps.consumerphotoeditor.fragments.ConsumerPhotoEditorActivity";
 
-        UiObject save = getUiObjectByText("SAVE", "android.widget.TextView");
-        save.click();
+        while (it.hasNext()) {
+            Map.Entry<String, SeekBarTestParams> pair = it.next();
+            Position pos = pair.getValue().seekBarPosition;
+            int steps = pair.getValue().steps;
+            int percent = pair.getValue().percent;
 
-        // Return to application home screen
-        getUiDevice().pressBack();
+            String runName = String.format(TAG + "_" + testTag + "_" + pair.getKey());
+            String gfxInfologName =  String.format(runName + "_gfxInfo.log");
+            String surfFlingerlogName =  String.format(runName + "_surfFlinger.log");
+
+            startDumpsysGfxInfo(parameters);
+            startDumpsysSurfaceFlinger(parameters, viewName);
+
+            Timer result = new Timer();
+            result = seekBarTest(seekBar, pos, steps);
+
+            stopDumpsysSurfaceFlinger(parameters, viewName, surfFlingerlogName);
+            stopDumpsysGfxInfo(parameters, gfxInfologName);
+
+            timingResults.put(runName, result);
+        }
+
+        saveAndReturn();
+    }
+
+    private void cropPhotoTest() throws Exception {
+        String testTag = "crop_photo";
+
+        // To improve travel accuracy perform the slide bar operation slowly
+        final int steps = 500;
```

Comment:
I've seen the travel distance vary occasionally if number of steps is low. Even a 400 for example.